extensions/jms/src/main/java/org/mule/extensions/jms/api/source/JmsListener.java

```diff
@@ -318,9 +288,25 @@ private boolean replyDestinationIsTopic(Destination destination) {
     return destination instanceof Topic;
   }
 
-  private void waitForMessageToBeProcesed() {
-    if (synchronous) {
-      jmsLock.lock();
+  private JmsListenerLock createJmsLock() {
+    return synchronous ? new DefaultJmsListenerLock() : new NullJmsListenerLock();
+  }
+
+  private int getValidNumberOfConsumers(int numberOfConsumers) {
+    if (numberOfConsumers > 1 && consumerType.isTopic()) {
+      TopicConsumer topicConsumer = (TopicConsumer) consumerType;
+
+      if (!isCapableOfMultiConsumersOnTopic(topicConsumer)) {
+        LOGGER.warn("Destination [" + destination + "] is a topic, but " + numberOfConsumers
```

Comment:
> Isn't it possible to have the destination as a subtype so that when it's topic there's no option to set consumers?
No, because in JMS 2 with a shared connection is possible to have multiple consumers for topics